### PR TITLE
[feature] connect codemirror to server

### PR DESCRIPTION
**Pull request by jtamsut**
You need to comment out line 1 in utilities.js to make the changes that were made to work.

### CODE REVIEW

File: client/components/Body.js [hunk 1]
```diff
@@ -5,13 +5,29 @@
 var React = require('react');
 var CodeMirror = require('./CodeMirror');
 var Analysis = require('./Analysis');
+var $ = require('jquery');
```
Comment (ghost) [inline client/components/Body.js]: move this to the line below requiring React (should move to line 6)

File: client/components/Body.js [hunk 2]
```diff
@@ -5,13 +5,29 @@
...
+      dataType: 'json',
+      contentType: 'application/json',
+      data: JSON.stringify({data: newCode}),
+      success: function() {
+        console.log('Post request was successful');
+      },
+      error: function(err) {
+        console.error('Ooop! You have a ' + err.status + ' error.');
+      }
+    })
```
Comment (ghost) [inline client/components/Body.js]: add a semicolon here, at the end of the ajax function (line 24)

File: client/components/Body.js [hunk 3]
```diff
@@ -5,13 +5,29 @@
...
+  },
   render: function() {
```
Comment (ghost) [inline client/components/Body.js:27]: insert a blank line between lines 25 and 26 (separating the getCode and render functions)

File: client/components/CodeMirror.js [hunk 4]
```diff
@@ -14,6 +14,9 @@ var CodeMirror = React.createClass({
       mode: 'javascript'
     }
   },
+  handleClick: function() {
+    this.props.getCode(this.state.code);
+  },
   updateCode: function(newCode) {
```
Comment (ghost) [inline client/components/CodeMirror.js:21]: insert blank line between lines 19 and 20 here

File: client/components/CodeMirror.js [hunk 5]
```diff
@@ -14,6 +14,9 @@ var CodeMirror = React.createClass({
...
     this.setState({
       code: newCode
```
Comment (ghost) [inline client/components/CodeMirror.js:23]: on line 23, at the end of the call `this.setState()` add a semicolon

Comment (ghost) [inline client/components/CodeMirror.js:23]: in addition, add a blank line between lines 24 and 25

File: client/components/CodeMirror.js [hunk 6]
```diff
@@ -28,6 +31,7 @@ var CodeMirror = React.createClass({
     return (
```
Comment (ghost) [inline client/components/CodeMirror.js:34]: insert a blank line right above the return to separate the variable declaration from it (between lines 30 and 31)

File: server/middleware.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 var utils = require('./utilities.js');
 
 module.exports.evalForAllInputSizes = function(req, res, next) {
-  var userInput = req.body.userInput;
-  
+  var userInput = req.body.data;
+
   console.log('M6-string received from ajax: ' + userInput);
```
Comment (mmurkidjanian) [inline server/middleware.js]: in the console.log, you need to change userInput to data

File: server/server.js [hunk 8]
```diff
@@ -4,15 +4,18 @@ var webpack = require('webpack');
 var webpackMiddleware = require('webpack-dev-middleware');
 var config = require('../webpack.config.js');
 var mid = require('./middleware.js');
+var bodyParser = require('body-parser');
```
Comment (ghost) [inline server/server.js]: move this up to line 3, so that it comes right after requiring express and right before requiring webpack

**Pull request merged**